src/editor/esphome-editor.ts

```diff
@@ -191,16 +286,23 @@ export class ESPHomeEditor extends LitElement {
     });
   }
 
-  static styles = css`
-    :host {
-      --editor-width: 100%;
-      --editor-height: 70vh;
-    }
-    main {
-      width: var(--editor-width);
-      height: var(--editor-height);
-    }
-  `;
+  calcEditorSize() {
+    return {
+      width: document.body.offsetWidth,
+      height: window.innerHeight - this.editor_header.offsetHeight,
+    };
+  }
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("resize", this._handleResize);
+  }
+  disconnectedCallback() {
+    window.removeEventListener("resize", this._handleResize);
+    super.disconnectedCallback();
+  }
+  _handleResize() {
```

Comment:
This needs to be a bound method or else `this` doesn't refer to this element
```suggestion
  _handleResize = () => {
```